backend/portal/api/curation/v1/curation/collections/collection_id/datasets/dataset_id/actions.py

```diff
@@ -37,7 +37,10 @@
 def get(collection_id: str, dataset_id: str = None):
     business_logic = get_business_logic()
 
+    # Double lookup
     collection_version = business_logic.get_collection_version_from_canonical(CollectionId(collection_id))
```

Comment:
nit: can we use the common util method get_infered_collection_version_else_forbidden? ignore if we want to return a 404 instead

Reply:
I think we need to refactor those methods but for now let's keep the existing behavior